theano/tensor/nnet/abstract_conv.py

```diff
@@ -109,7 +109,16 @@ def get_conv_shape_1axis(image_shape, kernel_shape,
         pad = border_mode
         if pad < 0:
             raise ValueError("border_mode must be >= 0")
-    out_shp = (image_shape + 2 * pad - kernel_shape) // subsample + 1
+
+            # In case of symbolic shape, we want to build the smallest graph
```

Comment:
It speed things up as we will build a smaller graph. If pad is 0, it is useless to add 2*0 in the graph. If subsample is 1, there is no need to divide by 1.

Mostly, this apply a few elemwise optimization manually.